src/interface/barplot.jl

```diff
@@ -12,44 +11,32 @@ using a dictionary `dict`. In that case, the keys will be used as
 the names and the values, which have to be numeric, will be used
 as the heights of the bars.
 
-Usage
-======
-
-    barplot(text, heights; xscale = :identity, title = "", xlabel = "", ylabel = "", labels = true, border = :barplot, margin = 3, padding = 1, color = :green, width = out_stream_width(out_stream), symbols = ['■'])
+# Usage
+    
+    barplot(text, heights; $(keywords((border = :barplot, color = :green,), remove = (:xlim, :ylim, :xscale, :yscale, :height, :grid), add = (:symbols,))))
 
     barplot(dict; kwargs...)
 
-Arguments
-==========
-
-- **`text`** : The labels / captions of the bars.
-
-- **`heights`** : The values / heights of the bars.
-
-- **`dict`** : A dictonary in which the keys will be used
-  as `text` and the values will be utilized as `heights`.
-
-- **`xscale`** : Function or Symbol to transform the bar length before plotting.
-  This effectively scales the x-axis without influencing the captions
-  of the individual bars. e.g. use `xscale = :log10` for logscale.
-
-$DOC_PLOT_PARAMS
+# Arguments
 
-- **`symbols`** : Specifies the characters that should be used to
-  render the bars
+$(arguments(
```

Comment:
It's better to leave a few comments or docstrings for the internal `keywords` and `arguments`. Otherwise, future contributors would easily get confused when they see a long string interpolation like this.

Reply:
Thanks, I've added the docstring, and even better now interpolate default kwargs into description and into concrete function arguments, so that changing a default value is now dead simple.